code/modules/mob/language.dm

```diff
@@ -287,10 +293,30 @@
 	flags = RESTRICTED | HIVEMIND
 
 /datum/language/grey/broadcast(mob/living/speaker, message, speaker_mask)
+	var/their = "their"
+	if(speaker.gender == "female")
+		their = "her"
+	if(speaker.gender == "male")
+		their = "his"
+	for(var/mob/living/player in view(4, speaker))
```

Comment:
Wasn't sure if 'visible_message' let you display in only a certain radius, and showing it off to everyone in sight might get rather annoying.